web/app/js/components/Sidebar.jsx

```diff
@@ -202,7 +218,66 @@ class Sidebar extends React.Component {
               <Menu.Item><PrefixedLink to="/pods">Pods</PrefixedLink></Menu.Item>
               <Menu.Item><PrefixedLink to="/replicationcontrollers">Replication Controllers</PrefixedLink></Menu.Item>
             </Menu.SubMenu>
+            <Menu.Divider />
+          </Menu>
 
+
+          <Menu
+            className="sidebar-menu"
+            theme="dark"
+            mode="inline"
+            selectedKeys={[normalizedPath]}>
+            { this.state.collapsed ? null : (
+              <Menu.Item className="sidebar-menu-item" key="/namespace-selector">
+                <Form layout="inline">
+                  <Form.Item>
+                    <Select
+                      defaultValue="All Namespaces"
+                      dropdownMatchSelectWidth={true}
+                      onChange={this.handleNamespaceSelector}>
+                      {
+                      _.map(namespaces, label => {
+                        return (
+                          <Select.Option key={label.value} value={label.value}>{label.name}</Select.Option>
+                        );
+                      })
+                    }
+                    </Select>
+                  </Form.Item>
+                </Form>
+              </Menu.Item>
+            )}
+
+            {this.state.collapsed ? null :
+              _.map(_.keys(sidebarComponents).sort(), resourceName => {
+                return (
+                  <Menu.SubMenu
+                    className="sidebar-menu-item"
+                    key={resourceName}
+                    disabled={sidebarComponents[resourceName].length === 0}
+                    title={<span>{friendlyTitle(resourceName).plural}</span>}>
+                    {
+                      _.map(_.sortBy(sidebarComponents[resourceName], r => `${r.namespace}/${r.name}`), r => {
+                        // only display resources that have been meshed
+                        if (r.added) {
```

Comment:
do we need this `if` here if we've filtered out the unadded components in `filterResourcesByNamespace`?